state/pgstatestorage.go

```diff
@@ -749,6 +767,42 @@ func (p *PostgresStorage) closeBatch(ctx context.Context, receipt ProcessingRece
 	return err
 }
 
+// UpdateGERInOpenBatch update ger in open batch
+func (p *PostgresStorage) UpdateGERInOpenBatch(ctx context.Context, ger common.Hash, dbTx pgx.Tx) error {
+	if dbTx == nil {
+		return ErrDBTxNil
+	}
+
+	var (
+		batchNumber   uint64
+		isBatchHasTxs bool
+	)
+	e := p.getExecQuerier(dbTx)
+	err := e.QueryRow(ctx, getLastBatchNumberSQL).Scan(&batchNumber)
+	if errors.Is(err, pgx.ErrNoRows) {
+		return ErrStateNotSynchronized
+	}
+
+	const isBatchHasTxsQuery = `SELECT EXISTS (SELECT 1 FROM state.batch WHERE raw_txs_data IS NOT NULL AND batch_num = $1)`
```

Comment:
`raw_txs_data` is only set once the batch is closed. The correct way of checking this is to check if there are l2blocks with the same batch number